transports/sqs/delayed-delivery.md

```diff
@@ -0,0 +1,146 @@
+
+```mermaid
+sequenceDiagram
+    participant S as Sender
+    participant D as Destination
+    participant F as Destination-delay.fifo
+    alt delay <= 900sec
+        S ->> D: Message DelaySecond = delay
+    else delay > 900sec
+        S -->> F:  Attribute DelaySecond = delay
+        loop every 900sec
+           alt remaining delay <= 900sec
+              F ->> D: Message DelaySecond = remaining delay
+           else remaining delay> 900sec
+              F -->> F:  Attribute DelaySecond = remaining delay
+           end
+        end
+end
+```
+
+### Clock drift
+
+In order to avoid clock drift the broker timestamps are used whereever possible to calculate the remaining timeout. The due time calculation uses `SentTimestamp` as well as `ApproximateFirstReceiveTimestamp` set by the broker. Only in cases of redelivery when `ApproximateReceiveCount` is higher than one the clients clock is used and thus subjected to clock drift.
+
+### Delivery
+
+For unrestricted delayed deliveries the last step is always a handover from the FIFO queue to the endpoint's input queue. SQS does not provide cross queue operation transactions thus the handover is subjected to retries. In cases of retries it might be possible that timeouts are delivered more than once. Message handlers need to be idempotent when used with transports with [transaction](transports/transactions.md) level `Receive Only` or below. The following diagram illustrates that:
+
+```mermaid
+sequenceDiagram
+    participant D as Destination
+    participant F as Destination-delay.fifo
+    F ->>+ F: Timeout due
+    F ->> D: Send with remaing delay
+    Note left of D: Original
+    F ->>- F: Delete Delayed Message failed
+    Note right of F: i.ex. network outage
+    F ->>+ F: Timeout due
+    F ->> D: Send with remaing delay
+    F ->>- F: Delete Delayed Message
+    Note left of D: Duplicate
+```
+
+### Example
+
+Below an example with a delayed delivery less or equal of 15 min.
+
+```mermaid
+graph LR
+
+subgraph Example: delay of 14 min and 5 seconds
+
+sender
+destination
+
+sender .-> |T1: Delay with 845sec| destination
+destination --> |"T2: fa:fa-hourglass-half 845sec"| destination
+
+end
+```
+
+14 min and 4 seconds is in total 845 seconds. This is less than 900 seconds and thus will be directly delayed to the destination with a `DelaySeconds` value of 845 seconds. No message attribute header will be used.
+
+Below an example with a delayed delivery greater than 15 min.
+
+```mermaid
+graph LR
```

Comment:
![image](https://user-images.githubusercontent.com/174258/37403363-e10f3792-278e-11e8-8118-747326e2de3e.png)

Reply:
Online editor doesn't render font-awesome. fa:fa-hourglass-half should be properly rendered on the docs site